### PR TITLE
fix(version): use tagged version instead of VERSION file while build

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does?**:
This PR removes the version file dependency on the image builds. This PR modifies the build scripts to take Travis tag or the version from the VERSION file of the build. This PR also refactors some functions that relied on the version to be specifically of the type `x.y.z` as the versions can differ with custom tags and RC tags. PR also comment outs some test cases based on the changes.

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:
After building images with a custom tag 1.11.0-ce-rc2
- Provisioned cstor SPC
- Provisioned cstor volume
- Provisioned jiva volume
- Deleted jiva volume
- Deleted cstor volume
- Deleted cstor SPC
- Deleted runtasks and restarted api-server, all the runtasks came back with correct version.
- Tested upgrade from 1.10.0

Runtasks with custom version
```sh
mayadata:setup$ kubectl get runtasks.openebs.io -n openebs
NAME                                                                  AGE
cas-volume-stats-default-1.11.0-ce-rc2                                    3m58s
cas-volume-stats-output-default-1.11.0-ce-rc2                             3m58s
cstor-pool-create-getspc-default-1.11.0-ce-rc2                            4m1s
cstor-pool-create-patchstoragepoolclaim-default-1.11.0-ce-rc2             4m1s
cstor-pool-create-putcstorpoolcr-default-1.11.0-ce-rc2                    4m1s
cstor-pool-create-putcstorpooldeployment-default-1.11.0-ce-rc2            4m1s
cstor-snapshot-create-createsnapshot-default-1.11.0-ce-rc2                3m59s
cstor-snapshot-create-listtargetservice-default-1.11.0-ce-rc2             3m59s
cstor-snapshot-create-output-default-1.11.0-ce-rc2                        3m59s
cstor-snapshot-delete-deletesnapshot-default-1.11.0-ce-rc2                3m58s
cstor-snapshot-delete-listtargetservice-default-1.11.0-ce-rc2             3m58s
cstor-snapshot-delete-output-default-1.11.0-ce-rc2                        3m58s
cstor-volume-create-getpvc-default-1.11.0-ce-rc2                          4m
cstor-volume-create-getstorageclass-default-1.11.0-ce-rc2                 4m
cstor-volume-create-listclonecstorvolumereplicacr-default-1.11.0-ce-rc2   4m
cstor-volume-create-listcstorpoolcr-default-1.11.0-ce-rc2                 4m
cstor-volume-create-output-default-1.11.0-ce-rc2                          4m
cstor-volume-create-putcstorvolumecr-default-1.11.0-ce-rc2                4m
cstor-volume-create-putcstorvolumereplicacr-default-1.11.0-ce-rc2         4m
cstor-volume-create-puttargetdeployment-default-1.11.0-ce-rc2             4m
cstor-volume-create-puttargetservice-default-1.11.0-ce-rc2                4m
cstor-volume-delete-deletecstorvolumecr-default-1.11.0-ce-rc2             3m59s
cstor-volume-delete-deletecstorvolumereplicacr-default-1.11.0-ce-rc2      3m59s
cstor-volume-delete-deletetargetdeployment-default-1.11.0-ce-rc2          3m59s
cstor-volume-delete-deletetargetservice-default-1.11.0-ce-rc2             3m59s
cstor-volume-delete-listcstorvolumecr-default-1.11.0-ce-rc2               3m59s
cstor-volume-delete-listcstorvolumereplicacr-default-1.11.0-ce-rc2        3m59s
cstor-volume-delete-listtargetdeployment-default-1.11.0-ce-rc2            3m59s
cstor-volume-delete-listtargetservice-default-1.11.0-ce-rc2               3m59s
cstor-volume-delete-output-default-1.11.0-ce-rc2                          3m59s
cstor-volume-list-listcstorvolumereplicacr-default-1.11.0-ce-rc2          4m
cstor-volume-list-listpv-default-1.11.0-ce-rc2                            4m
cstor-volume-list-listtargetpod-default-1.11.0-ce-rc2                     4m
cstor-volume-list-listtargetservice-default-1.11.0-ce-rc2                 4m
cstor-volume-list-output-default-1.11.0-ce-rc2                            3m59s
cstor-volume-read-listcstorvolumecr-default-1.11.0-ce-rc2                 3m59s
cstor-volume-read-listcstorvolumereplicacr-default-1.11.0-ce-rc2          3m59s
cstor-volume-read-listtargetpod-default-1.11.0-ce-rc2                     3m59s
cstor-volume-read-listtargetservice-default-1.11.0-ce-rc2                 3m59s
cstor-volume-read-output-default-1.11.0-ce-rc2                            3m59s
jiva-snapshot-create-invokehttp-default-1.11.0-ce-rc2                     3m58s
jiva-snapshot-create-listsourcetargetservice-default-1.11.0-ce-rc2        3m58s
jiva-snapshot-create-output-default-1.11.0-ce-rc2                         3m58s
jiva-snapshot-podsinopenebsns-default-1.11.0-ce-rc2                       3m58s
jiva-volume-create-getpvc-default-1.11.0-ce-rc2                           4m2s
jiva-volume-create-getstorageclass-default-1.11.0-ce-rc2                  4m2s
jiva-volume-create-getstoragepoolcr-default-1.11.0-ce-rc2                 4m2s
jiva-volume-create-output-default-1.11.0-ce-rc2                           4m2s
jiva-volume-create-putreplicadeployment-default-1.11.0-ce-rc2             4m2s
jiva-volume-create-puttargetdeployment-default-1.11.0-ce-rc2              4m2s
jiva-volume-create-puttargetservice-default-1.11.0-ce-rc2                 4m2s
jiva-volume-delete-deletereplicadeployment-default-1.11.0-ce-rc2          4m1s
jiva-volume-delete-deletetargetdeployment-default-1.11.0-ce-rc2           4m1s
jiva-volume-delete-deletetargetservice-default-1.11.0-ce-rc2              4m1s
jiva-volume-delete-listreplicadeployment-default-1.11.0-ce-rc2            4m1s
jiva-volume-delete-listreplicapod-default-1.11.0-ce-rc2                   4m1s
jiva-volume-delete-listtargetdeployment-default-1.11.0-ce-rc2             4m1s
jiva-volume-delete-listtargetservice-default-1.11.0-ce-rc2                4m1s
jiva-volume-delete-output-default-1.11.0-ce-rc2                           4m1s
jiva-volume-delete-putreplicascrub-default-1.11.0-ce-rc2                  4m1s
jiva-volume-isvalidversion-default-1.11.0-ce-rc2                          4m4s
jiva-volume-list-listpv-default-1.11.0-ce-rc2                             4m3s
jiva-volume-list-listreplicapod-default-1.11.0-ce-rc2                     4m3s
jiva-volume-list-listtargetpod-default-1.11.0-ce-rc2                      4m3s
jiva-volume-list-listtargetservice-default-1.11.0-ce-rc2                  4m3s
jiva-volume-list-output-default-1.11.0-ce-rc2                             4m3s
jiva-volume-podsinopenebsns-default-1.11.0-ce-rc2                         4m4s
jiva-volume-read-listreplicapod-default-1.11.0-ce-rc2                     4m2s
jiva-volume-read-listtargetpod-default-1.11.0-ce-rc2                      4m3s
jiva-volume-read-listtargetservice-default-1.11.0-ce-rc2                  4m3s
jiva-volume-read-output-default-1.11.0-ce-rc2                             4m2s
jiva-volume-read-patchreplicadeployment-default-1.11.0-ce-rc2             4m2s
jiva-volume-read-verifyreplicationfactor-default-1.11.0-ce-rc2            4m2s
storage-pool-list-default-1.11.0-ce-rc2                                   3m58s
storage-pool-list-output-default-1.11.0-ce-rc2                            3m58s
storage-pool-read-default-1.11.0-ce-rc2                                   3m58s
storage-pool-read-output-default-1.11.0-ce-rc2                            3m58s
```

**Any additional information for your reviewer?** : 
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: pkg/util/unstructured.go [hunk 1]
```diff
@@ -28,12 +28,15 @@ package util
...
 	"fmt"
-	"k8s.io/apimachinery/pkg/util/json"
 	"strings"
+
+	"k8s.io/apimachinery/pkg/util/json"
 )
 
 // GetNestedField returns a nested field from the provided map
 func GetNestedField(obj map[string]interface{}, fields ...string) interface{} {
+	fmt.Println("obj", obj)
```
Comment (mittachaitu) [inline pkg/util/unstructured.go:38]: remove this debug

File: pkg/apis/openebs.io/v1alpha1/versionDetails.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ var (
 		"1.8.0": true, "1.9.0": true, "1.10.0": true, "1.11.0": true,
 		"1.12.0": true,
 	}
-	validDesiredVersion = version.GetVersion()
+	validDesiredVersion = strings.Split(version.GetVersion(), "-")[0]
```
Comment (mittachaitu) [inline pkg/apis/openebs.io/v1alpha1/versionDetails.go:33]: Q: When we perform an upgrade of the OpenEBS version will still deleting old version runtask and CASTemplates?

Comment (shubham14bajpai) [reply pkg/apis/openebs.io/v1alpha1/versionDetails.go:33]: Yes on upgrade the old runtasks and castemplates are cleaned.

**Comment by mittachaitu** [comment]
`make test` is failing in Travis we can change the expected output to small case letters get test case fixed(since we changed the behavior of function).

**Pull request merged**